backend/cmm_helpers.ml

```diff
@@ -2203,25 +2203,27 @@ let if_operation_supported_bi bi op ~f =
   if bi = Primitive.Pint64 && size_int = 4 then None
   else if_operation_supported op ~f
 
-let clz bi arg dbg =
-  let op = Cclz { arg_is_non_zero = false; } in
+let clz ~arg_is_non_zero bi arg dbg =
+  let op = Cclz { arg_is_non_zero; } in
   if_operation_supported_bi bi op ~f:(fun () ->
     let res = Cop(op, [make_unsigned_int bi arg dbg], dbg) in
     if bi = Primitive.Pint32 && size_int = 8 then
       Cop(Caddi, [res; Cconst_int (-32, dbg)], dbg)
     else
       res)
 
-let ctz bi arg dbg =
+let ctz ~arg_is_non_zero bi arg dbg =
   let arg = make_unsigned_int bi arg dbg in
   if bi = Primitive.Pint32 && size_int = 8 then begin
+    (* regardless of the value of the argument [arg_is_non_zero],
+       always set the corresponding field to [true]. *)
```

Comment:
This comment just says what the code says.  Let's change it to say what the reason is...